docs/howto/write-unit-tests-for-a-charm.md

```diff
@@ -161,3 +161,23 @@ def test_charm_runs(my_charm):
 
 If you use pytest, you should put the `my_charm` fixture in a top level `conftest.py`, as it will likely be shared between all your unit tests.
 ```
+
+## Test a function that receives charm as an argument
+
+A function might take in charm as an argument to extract charm state or charm relation information. To test such functions, you can use
+the following method to test the function.
+
+
+```python
+# Example function to test that takes the charm as an argument
+def get_num_relation(my_charm):
+    return len(self.model.relations.get("my-relation", []))
+
+# Test for the function that takes the charm as an argument
+def test_get_num_relation():
+    ctx = testing.Context(my_charm)
+    state_in = testing.State(relations=[])
+    with ctx(ctx.on.update_status(), state_in) as mgr: # use any hook event here
+        result = takes_charm_as_argument(my_charm=mgr.charm)
```

Comment:
Here, it's unclear why the reference to the charm class is being gotten from `mgr.charm` when it's available as `my_charm` at the start of test function body.

How about this example is split into two functions:
- a pytest.fixture that internally uses the charm type and returns a Context ready to be used
- a test function that only receives the Context and thus has a reason to look up the charm type through the Context or Manager.

Reply:
> Which also means we'd have to document the exact "point in time" or charm object state that the test would find mgr.charm in.

That would be great - the purpose of this recipe is to just take the "state" of the charm (integrations, storages, configurations, ...etc) and see if a smaller handler for a particular "state" is able to pick up the state as intended and is able to operate on it.